samza-core/src/main/java/org/apache/samza/zk/ZkUtils.java

```diff
@@ -324,4 +398,45 @@ public void subscribeToProcessorChange(IZkChildListener listener) {
     zkClient.subscribeChildChanges(keyBuilder.getProcessorsPath(), listener);
     metrics.subscriptions.inc();
   }
+
+  /**
+   * Represents zookeeper processor node.
+   */
+  private static class ProcessorNode {
```

Comment:
Reasons: 

* This is a `DAO` and mimics the exact way how processors and it's data are organized in zookeeper. There's a proper logical mapping between this to how it exists in zookeeper. I think ephemeralPath doesn't belong to ProcessorData. In future when the codebase evolves, will help readability.

* ProcessorNode is a place for storing creationTime, numberOfChildren and other statistics along with ACL related info in future.

Makes sense? Please share your thoughts.